### DISCUSSION TITLE
non-deterministic build artifacts - content hash (v13)

### DISCUSSION THREAD

**Discussion by steve-haar** [vercel/next.js#50713] [Help]
### Summary

I'm using next v13.4.4 and `output: 'export'` option with a static `buildId`. When I perform multiple clean builds via `next build` all of my output files are the same except for my root `layout-{hash}.js` file. For each build this results in a different file hash. However, if I persist the `.next/cache/webpack/client-production` folder, then my builds will be deterministic.

My expectation is that the content hash of the file should remain the same if none of the source files have changed. My current build pipelines requires that these builds be deterministic. I've seen this problem before with NextJS, but it looks like this should have been resolved. Is this expected or is there any way to get consistent content hashes across builds?

**Comment by barroudjo** [comment]
I have the same problem except it goes further: even when persisting the `.next` folder my build (with `output: 'export'` with a static buildId) is non deterministic. By that I mean that for some files the hash in their filename changes. And when I look at the detail of the changes, it turns out it's only shuffling code around, not real changes.

This is highly problematic I think... Builds should be deterministic. In our case it means that when deploy new content from our CMS (so absolutely no code changes) all users returning on our site have to redownload the exact same CSS and JS, only because the hashes have changed because some code, for no good reason, has been shuffled around...

Note: this is with Next.js 14.1.2, on linux, on a bunch of different machines and gitlab CI

**Reply by MaxLiebsch** [reply]
Having the similar behaviour with 14.3.2 but with a different problem. The same unchanged code sometimes builds sometimes it does not and throws errors and I cannot see the pattern behind it.
